docs/design/architectural-overview.md

```diff
@@ -0,0 +1,168 @@
+
+It is the OCM workagent that is responsible for syncing down and applying the resources into the managed spoke cluster. It is also responsible for syncing status information back to the hub. 
+
+The status information reported back to the Hub is used by the multi-cluster gateway controller to know what LB hosts / IPAddresses to use for DNSRecords that it creates and manages.
+
+More info can be found here https://open-cluster-management.io/concepts/architecture/ 
+
+## How does Kuadrant integrate with Gateway Providers
+
+The Kuadrant data plane, integrates with the Istio gateway provider only currently: 
+
+- It registers Authorino with the `IstioOperator` as an auth provider.
+
+- It leverages an `EnvoyFilter`  to register the rate limiting service as a upstream service.
+
+- It leverages istio `AuthorizationPolicy` to configure when Authorino should be called for a given host and path etc. 
+
+- It provides a WASM Plugin that conforms to the Proxy WASM ABI. This WASM Plugin is configured via the Kuadrant Operator based on valid rate limit policies and is then executed by the underlying Envoy instance that provides the proxy layer for the Gateway (IE Envoy). This plugin currently manages when RateLimiting and knows the location of the Limitador service.
+
+
+
+
+### Data Flows
+
+There are several different data flows when using Kuadrant. 
+
+**Control plane configuration and status reporting**
+
+The initial creation of these APIs (gateways, policies etc) is done by the relevant persona in the control plane just as they would any other k8s resource. We use the term cluster admin or gateway admin as the operatoins type persona configuring, and placing gateways. 
+As shown above, in a multi-cluster configuration. API definitions are pulled from the Hub and "manifested" into the spokes. The Status of those synced resources are reported back to the Hub. The same happens for a single cluster, the only difference being the work agent hub controllers are all installed on one cluster.
+
+**3rd party enforcement and Integration**
+
+In order to enforce the policy configuration, components in the control plane and data plane can reach out to configured 3rd parties such as cloud based DNS provider, TLS providers and Auth providers.
+
+
+**Request Flow**
+
+Requests coming through the gateway instance are intercepted by the WASM plugin installed into the Envoy based gateway provider. This plugin is then responsible for passing on the request to Limitador to be rate limited or Authorino to ensure auth. Each of these component each have the capability to see the request and need to in order to make the required decision. Each of these component can also prevent the request from reaching its backend destination based on user configuration. 
+
+
+![](./images/request-flow.png)
+
+
+## Auth
+
+Authentication is handled by the underly kubernetes cluster. All APIs are CRDs and so authorization is controlled by RBAC in the same way as any kubernetes API.
```

Comment:
Worth elaborating this a bit further perhaps? To make it clear it is about authenticating/authorising for using Kuadrant (as an App developer, as a cluster operator), and nothing to do with the "auth" Kuadrant delivers as a feature in the data plane. WDYT?